### ISSUE TITLE
JAWS virtual cursor is not auto-activated in Console Output region

### ISSUE THREAD

**Issue by jooyoungseo** [rstudio/rstudio#6884]
### System details

    RStudio Edition : Server
    RStudio Version : 1.3.958
    OS Version      : Windows 10 and JAWS 2020 in Chrome
    R Version       : any

### Steps to reproduce the problem

1. With JAWS loaded, open RStudio Server in Chrome.
2. Make sure to enable the following settings under `Help / Accessibility / Accessibility Options...`

- [x] Screen reader support (requires restart)
- [x] Entire page has application role (requires restart)

3. Press `Ctrl+2` to focus in the console.
4. Type `summary(airquality$Ozone)` in the console and hit enter.
5. Press either `Alt+Shift+2` or `Shift+Tab` to focus in "Console Output" area.
6. Press up/down-arrow keys to see if you would hear anything in the console output.

### Describe the problem in detail

When landed in "Console Output" region, JAWS does not auto-switch to virtual cursor unlike NVDA.

This means users have to manually turn back on the virtual cursor by themselves to read the content in the console output.

Users could get easily confused and distracted by this mode-transition behavior.

### Describe the behavior you expected

JAWS must auto-switch to virtual cursor in "Console Output" area denoted with `role = "document"`.

- [X] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [X] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [X] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by jooyoungseo** [comment]
Alright, I think I have figured this out.

To cut to the chase, this is a JAWS bug that it does not inherit "document" role from the parent container when focused on the child element.

I have made the following minimal example to repro this issue:

``` html
<div id = "main" role = "application">
  <div id = "output" role="document" tabindex = "0">
    <div id = "output-content" accesskey="o" tabindex="-1">
      <h1>The Life of Albert Einstein</h1>
      <p>Lorem ipsum...</p>
    </div>
  </div>

  <label for="test-textarea">Test Text Area</label>
  <textarea accesskey="t" id="test-textarea" rows="4" cols="50">
    This is text area where you can type anything.
  </textarea>
</div>
```

JAWS does not activate the virtual cursor when focused on the "output-content" area just like our RStudio Server issue.

You can resolve this issue by re-assigning "document" role for the `id = "output-content"` div as it is not inherited from its parent div for JAWS.

In our case, please add `role = "document"` to the `<span>` element having "focus-visible" class under the `<pre>` container.

**Comment by jooyoungseo** [comment]
@gtritchie, would you mind considering this towards 1.3 patch as well? I think this buggy behavior coming from JAWS is critical and very inconenient for JAWS users. Any advice would be greatly appreciated.

**Comment by gtritchie** [comment]
Ported to 1.3-patch: 38be198d

**Comment by R-Hannibal** [comment]
Hey @jooyoungseo, fix is now in place in the 1.3 patch build (accessible on the [Preview Page](https://rstudio.com/products/rstudio/download/preview/) now).

If you get the chance will you please let us know if the issue is resolved for you in that build?

**Comment by jooyoungseo** [comment]
Hi @R-Hannibal! Gary and I have already confirmed the issue has been resolved in the patch.

**Comment by R-Hannibal** [comment]
Thanks @jooyoungseo!

**Comment by gtritchie** [comment]
Removing verified: I can confirm this fix is NOT in Water Lily build 1.3.1006.

**Comment by R-Hannibal** [comment]
@jooyoungseo - it's probably just my natural QA-paranoia speaking, but would you mind re-verifying this issue is still looking fixed in the latest preview build (available [HERE](https://rstudio.com/products/rstudio/download/preview/))? 

After some fixes failed to merge over successfully to the patch branch originally I want to make absolutely sure we're good before I close. Thank you for the assistance and please let me know if you're busy and I'll figure something else out :)

**Comment by jooyoungseo** [comment]
@R-Hannibal, I have just verified this issue has been resolved in v1.3.1023.
